python/kvikio/nvcomp.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+
+from enum import Enum
+
+import cupy as cp
+import numpy as np
+
+import kvikio._lib.pynvcomp as _lib
+
+
+def cp_to_nvcomp_dtype(in_type: cp.dtype) -> Enum:
```

Comment:
It was a coercion function until I added mypy notations. Now it is only called by `CascadedCompressor` and the type has already been coerced, so it just converts to the proper NVCOMP_TYPE. I'll drop the coercion.

Reply:
So, coming back to this I'm reminded that there are many representations of valid cp.dtypes. The coercion is because, for example, `cp.int8` defined during runtime is actually `<class 'numpy.int8'>`, which doesn't seem to be caught by `mypy` in this context. There are many more valid dtype specifications that won't be caught by `mypy` such as passing one of the string variations that should be permitted. I'm going back to the coercion form. `mypy` annotations are just suggestions, after all. :)